### DISCUSSION TITLE
Querying Database Views

### DISCUSSION THREAD

**Discussion by davidkroell** [strapi/strapi#8017] [Legacy Topics]
### TL DR 
Is there a way to query a database view (SELECT *) within a Strapi instance, model or something else?
**Error: ER_WRONG_OBJECT: 'strapi.vwAnalytics_1' is not BASE TABLE**

I am currently building a simple sale management application with Strapi and I am really happy about it. Thank you so far for your amazing work!

But now, I'd like to implement analytics in my Application (at the end of the day there should be charts on the UI).

I tried to manipulate my model to use a database view instead of a table, because I feel very confident using advanced SQL query techniques. 
This is my definition I'd like to use. `vwAnalytics_1` is the name of the view. It returns the two columns WeekNum and SalesPerWeek.
```json
{
  "kind": "collectionType",
  "connection": "default",
  "collectionName": "vwAnalytics_1",
  "info": {
    "name": "overview"
  },
  "options": {
    "comment": ""
  },
  "attributes": {
    "WeekNum": {
      "type": "int"
    },
    "SalesPerWeek": {
      "type": "int"
    }
  }
}
```

Error callstack on application startup:
```
> strapi "dev"

[2020-09-23T19:04:38.675Z] debug ⛔️ Server wasn't able to start properly.
[2020-09-23T19:04:38.677Z] error Error: ER_WRONG_OBJECT: 'strapi.vwAnalytics_1' is not BASE TABLE
    at Query.Sequence._packetToError (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\protocol\sequences\Sequence.js:47:14)
    at Query.ErrorPacket (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\protocol\sequences\Query.js:79:18)
    at Protocol._parsePacket (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\protocol\Protocol.js:291:23)
    at Parser._parsePacket (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\protocol\Parser.js:433:10)
    at Parser.write (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\protocol\Parser.js:43:10)
    at Protocol.write (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\protocol\Protocol.js:38:16)
    at Socket.<anonymous> (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\Connection.js:88:28)
    at Socket.<anonymous> (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\Connection.js:526:10)
    at Socket.emit (events.js:315:20)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:271:9)
    at Socket.Readable.push (_stream_readable.js:212:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:186:23)
    at Protocol._enqueue (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\protocol\Protocol.js:144:48)
    at Connection.query (C:\Users\david\Documents\zichnErp\node_modules\mysql\lib\Connection.js:198:25)
    at C:\Users\david\Documents\zichnErp\node_modules\knex\lib\dialects\mysql\index.js:130:18
```

If this would not work, is there another way to query a database view and return the results as JSON?

Best regards,
David

**Comment by derrickmehaffy** [comment]
I don't believe we currently support database views natively however you can directly hook into connectors to create custom queries or, in the case of SQL, you can directly access Knex and issue RAW SQL commands.

For custom queries you will want to check out the api route here, but you will want to refer to the Bookshelf or Mongoose documentation (depending on which database you are using): https://strapi.io/documentation/v3.x/concepts/queries.html#custom-queries

For RAW SQL queries, it's much simpler using: `strapi.connections.default.raw('Your SQL Statement here')`
